src/Raven.Server/Web/System/TcpConnectionInfoHandler.cs

```diff
@@ -19,5 +27,116 @@ public Task Get()
 
             return Task.CompletedTask;
         }
+
+        [RavenAction("/info/tcp/pull", "GET", AuthorizationStatus.UnauthenticatedClients)]
+        public async Task GetPullReplicationInfo()
+        {
+            var databaseName = GetStringQueryString("databaseName");
+            var databaseGroupId = GetStringQueryString("databaseGroupId");
+            var definitionName = GetStringQueryString("definitionName");
+
+            var nodes = GetResponsibleNode(databaseName, databaseGroupId, definitionName);
+
+            // TODO: cache the tcp info for each node?
```

Comment:
No, there is no need for that. The number of calls you are likely to get is minuscule.